projects/plugins/jetpack/extensions/blocks/videopress/edit.js

```diff
@@ -596,40 +601,84 @@ const VideoPressEdit = CoreVideoEdit =>
 				</Fragment>
 			);
 
+			const filesSelected = files => {
+				this.setState( { fileForUpload: files[ 0 ] } );
+			};
+
+			// Handle Media Library selection
+			const mediaItemSelected = item => {
+				if ( item && item.videopress_guid ) {
+					this.props.setAttributes( { guid: item.videopress_guid } );
+				} else {
+					this.props.setAttributes( { src: item.url } );
+				}
+			};
+
+			const uploadFinished = ( videoGuid = null ) => {
+				this.setState( { fileForUpload: null } );
+				if ( videoGuid ) {
+					setAttributes( { guid: videoGuid } );
+				}
+			};
+
+			const isResumableUploading = null !== fileForUpload && fileForUpload instanceof File;
+			if ( isResumableUploading ) {
+				return (
+					<VideoPressBlockProvider onUploadFinished={ uploadFinished }>
+						<Fragment>
+							{ blockSettings }
+							<ResumableUpload file={ fileForUpload } { ...this.props } />
+						</Fragment>
+					</VideoPressBlockProvider>
+				);
+			}
+
 			/*
 			 * The Loading/CoreVideoEdit blocks should be in the tree if :
 			 *     - We don't have a video GUID
 			 *     - Or we're uploading a video
 			 *     - Or we're in fallback mode (to display a video hosted locally for instance)
 			 * In all other cases, we should be able to safely display the Loading/VpBlock branch.
 			 */
+
 			const isFetchingVideo = isFetchingMedia || isFetchingPreview;
 			const renderCoreVideoAndLoadingBlocks = fallback || isUploading || ! guid;
 			const displayCoreVideoBlock =
 				renderCoreVideoAndLoadingBlocks && ! isUploading && ! isFetchingVideo;
 
 			// In order for the media placeholder to keep its state for error messages, we need to keep the CoreVideoEdit component in the tree during file uploads.
 			// Keep this section separate so the CoreVideoEdit stays in the tree, once we have a video, we don't need it anymore.
+			const coreVideoFragment = (
```

Comment:
Maybe the coreVideoFragment can be kept under the following if ( renderCoreVideoAndLoadingBlocks )